feg/gateway/services/session_proxy/credit_control/gx/definitions.go

```diff
@@ -133,6 +135,9 @@ type RuleDefinition struct {
 	RedirectInformation *RedirectInformation `avp:"Redirect-Information"`
 	Qos                 *QosInformation      `avp:"QoS-Information"`
 	ServiceIdentifier   *uint32              `avp:"Service-Identifier"`
+	Online           uint32                  `avp:"Online"`
+	Offline          uint32                  `avp:"Offline"`
```

Comment:
Online/Offline are Enumerated, enumerated data type is int32, not uint32, see: https://datatracker.ietf.org/doc/html/rfc3588

Reply:
Ya I checked that but all the other Enumerated fields on go-diameter use `uint32` or `uint8`

check `Usage-Monitoring-Level` or `Event-Trigger` 

https://github.com/fiorix/go-diameter/blob/master/diam/dict/testdata/gx_credit_control.xml#L149-L149

Is it possible go diameter just convert this into uints?

Reply:
it'll work with uints, but, I'd just make it consistent with the spec & go-diameter, see: https://github.com/fiorix/go-diameter/blob/master/diam/datatype/enum.go#L10 & https://github.com/fiorix/go-diameter/blob/master/diam/datatype/int32.go#L13